### PR TITLE
Nuget automatic version increment

### PR DESCRIPTION
**Pull request by rflechner**
Hi,

There is a small module I use to auto increment my nugets packages versions.

For example, we can get current published version of FAKE nuget with:
```fsharp
getlastNugetVersion (NugetVersionArg.Default().Server) "FAKE"
//val it : string option = Some "4.12.0"
```

To auto increment a nuget, we can do this:
```fsharp
NuGet (fun p -> 
            {p with
                 // Other settings truncated ....
                Project = "MyProject"
                Title = "MyProject"
                Version = nextVersion (fun arg -> { arg with PackageName="MyProject" })
             })
            "MyProject.nuspec"
```
 We can choose increment method with:
```fsharp
nextVersion (fun arg -> { arg with PackageName="FAKE"; Increment=IncBuild })
//val it : string = "4.12.1"

nextVersion (fun arg -> { arg with PackageName="FAKE"; Increment=IncMinor })
//val it : string = "4.13.0"

nextVersion (fun arg -> { arg with PackageName="FAKE"; Increment=IncMajor })
//val it : string = "5.12.0"
```

Regards

### CODE REVIEW

**Comment by forki** [comment]
I think this should use our SemVer model

**Comment by rflechner** [comment]
Of course, you're right, I didn't saw this model.
I replaced System.Version using by Fake.SemVerHelper.SemVerInfo

File: src/app/FakeLib/NuGet/NugetVersion.fs [hunk 1]
```diff
@@ -0,0 +1,74 @@
+﻿[<AutoOpen>]
```
Comment (forki) [inline src/app/FakeLib/NuGet/NugetVersion.fs]: please remove the AutoOpen

File: src/app/FakeLib/NuGet/NugetVersion.fs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+module Fake.NugetVersion
```
Comment (forki) [inline src/app/FakeLib/NuGet/NugetVersion.fs]: Please use "NuGet"

File: src/app/FakeLib/NuGet/NugetVersion.fs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    fun (v:SemVerInfo) ->
+        { v with Build=""; Patch=0; Minor=(v.Minor+1) }
+
+/// Increment major version
+let IncMajor:NugetVersionIncrement = 
+    fun (v:SemVerInfo) ->
+        { v with Build=""; Patch=0; Minor=0; Major=(v.Major+1) }
+
+/// Arguments for the next nuget version number computing
+type NugetVersionArg =
```
Comment (forki) [inline src/app/FakeLib/NuGet/NugetVersion.fs]: please always use "NuGet"

**Comment by rflechner** [comment]
Ok, I removed the AutoOpen and renamed nuget to NuGet.

File: src/app/FakeLib/FakeLib.fsproj [hunk 4]
```diff
@@ -139,6 +139,7 @@
     <Compile Include="NuGet\NugetHelper.fs" />
     <Compile Include="NuGet\Update.fs" />
     <Compile Include="NuGet\Install.fs" />
+    <Compile Include="NuGet\NugetVersion.fs" />
```
Comment (forki) [inline src/app/FakeLib/FakeLib.fsproj]: here too ;-)

File: src/app/FakeLib/NuGet/NugetVersion.fs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      DefaultVersion:string }
+    /// Default arguments to compute next NuGet version number
+    static member Default() =
+        { Server="https://www.nuget.org/api/v2"
+          PackageName=""
+          Increment=IncMinor
+          DefaultVersion="1.0" }
+
+/// Retrieve current NuGet version number
+let getlastNuGetVersion server (packageName:string) =
```
Comment (forki) [inline src/app/FakeLib/NuGet/NugetVersion.fs]: getLastNuGetVersion

Comment (forki) [inline src/app/FakeLib/NuGet/NugetVersion.fs]: sorry for being pedantic

Comment (rflechner) [inline src/app/FakeLib/NuGet/NugetVersion.fs]: no problem, thank you for the code review

**Pull request merged**